test/behaviours/code/insert-block.js

```diff
@@ -0,0 +1,16 @@
+/** @jsx h */
+
+import { Transforms } from 'slate'
+
+export default editor => Transforms.insertNodes(
+    editor, { type: 'code', placement: 'block', children: [{ text: '' }] })
```

Comment:
I believe we've discussed this somewhere else, so I'll put just a short summary here:

While `Code.insertCode(editor, 'block')` is indeed slightly shorter than `Transforms.insertNodes(editor, { type: 'code', placement: 'block', children: [{ text: '' }] })` (at 32 characters versus 94, or one line vs two when formatted), it also hides what actually happens. The intent behind (and outcome of) `Transform.insertNodes` is obvious to the reader, with `Code.insertCode` not so much. For this reason I've only made utility functions for operations which are not trivial.

Reply:
I was thinking about creating such functions because it would be easier to refactor code in the future if something would change in the slate's core. Currently, we would have to update `cnx-designer` and `adaptarr-front` and if `adaptarr-front` would use functions like `Code.insertCode(...)`, `Code.wrapAsCode`, etc. we would have to update only `cnx-designer` which would be a lot simpler and finally allow us to separate front-end and editor's backend

Reply:
https://github.com/openstax-poland/cnx-designer/issues/93